core/Pos/Util/Modifier.hs

```diff
@@ -141,9 +145,13 @@ insertionsMap = HM.mapMaybe identity . getMapModifier
 insertions :: MapModifier k v -> [(k, v)]
 insertions = HM.toList . insertionsMap
 
+-- | Get all deletions from 'MapModifier' as a 'HashSet'.
+deletionsSet :: MapModifier k v -> HashSet k
+deletionsSet = getKeys . HM.filter isNothing . getMapModifier
+
 -- | Get all deletions from 'MapModifier'.
 deletions :: MapModifier k v -> [k]
-deletions = HM.keys . HM.filter isNothing . getMapModifier
+deletions = HS.toList . deletionsSet
```

Comment:
It's a bit less efficient, because it involves conversion to set and then to list, but ok.